test/cbmc/include/core_sntp_stubs.h

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * @brief Application defined function to generate and append
+ * authentication code in an SNTP request buffer for the SNTP client to be
+ * authenticated by the time server, if a security mechanism is used.
+ *
+ * @param[in] pContext Application defined authentication interface context.
+ * @param[in] pTimeServer The time server being used to request time from.
+ * This parameter is useful to choose the security mechanism when multiple time
+ * servers are configured in the library, and they require different security
+ * mechanisms or authentication credentials to use.
+ * @param[in] pBuffer SNTP request buffer.
+ * @param[in] bufferSize The maximum amount of data that can be held by the buffer.
+ * @param[out] pAuthCodeSize This should be filled with size of the authentication
+ * data appended to the SNTP request buffer, @p pBuffer.
+ *
+ * @return The function SHOULD return one of the following integer codes:
+ * - #SntpSuccess when the authentication data is successfully appended to @p pBuffer.
+ * - #SntpErrorBufferTooSmall when the user-supplied buffer (to the SntpContext_t through
+ * @ref Sntp_Init) is not large enough to hold authentication data.
+ */
+SntpStatus_t GenerateClientAuthStub( SntpAuthContext_t * pContext,
+                                     const SntpServerInfo_t * pTimeServer,
+                                     void * pBuffer,
+                                     size_t bufferSize,
+                                     size_t * pAuthCodeSize );
+
+/**
+ * @brief Application defined function to resolve time server domain-name
+ * to an IPv4 address.
+ *
+ * @param[in] pTimeServer The time-server whose IPv4 address is to be resolved.
+ * @param[out] pIpV4Addr This should be filled with the resolved IPv4 address.
+ * of @p pTimeServer.
+ *
+ * @return `true` if DNS resolution is successful; otherwise `false` to represent
+ * failure.
+ */
+bool ResolveDnsFuncStub( const SntpServerInfo_t * pServerAddr,
+                         uint32_t * pIpV4Addr );
+
+/**
+ * @brief Application defined function to obtain the current system time
+ * in SNTP timestamp format.
+ *
+ * @param[out] pCurrentTime This should be filled with the current system time
+ * in SNTP timestamp format.
+ *
+ * @return `true` if obtaining system time is successful; otherwise `false` to
+ * represent failure.
+ */
+bool GetTimeFuncStub( SntpTimestamp_t * pCurrentTime );
+
+/**
+ * @brief Application defined function to update the system clock time
+ * so that it is synchronized the time server used for getting current time.
+ *
+ * @param[in] pTimeServer The time server used to request time.
+ * @param[in] pServerTime The current time returned by the @p pTimeServer.
+ * @param[in] clockOffSetSec The calculated clock offset of the system relative
+ * to the server time.
+ *
+ * @return `true` if setting system time is successful; otherwise `false` to
+ * represent failure.
+ */
+bool SetTimeFuncStub( const SntpServerInfo_t * pTimeServer,
```

Comment:
The definition is not correct, the `SntpSetTime_t` function does not return a `bool` type, but `void` type.